### DISCUSSION TITLE
Picture element with q-img

### DISCUSSION THREAD

**Discussion by Divensky** [quasarframework/quasar#17308] [General - Components / Directives / etc]
I have a mobile image and a desktop image and I want to display the one that fits the window width. This is an [example of what I want to achieve, based on MDN](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images#art_direction):
```
<picture>
  <source media="(max-width: 799px)" srcset="mobile.jpg" />
  <source media="(min-width: 800px)" srcset="desktop.jpg" />
  <img src="desktop.jpg"/>
</picture>
```
I searched for ways to do it and found the following: 

- I cannot use `q-img` inside the `<picture>` element because the `<picture>` element requires an `img` tag as its direct child. 
- No `q-img` props permit me to achieve a functionality exactly like that of the `<picture>` element above.
- I could use a dynamic path in the `q-img` and calculate this path either by observing the window size directly or though `$q.screen`; however, either calculation would introduce a performance overhead which I don't want. 

So I concluded that the way to display responsive images at alternative dynamic paths without a performance penalty would be by using native `<picture>` and `<img>` elements rather than `q-img`.

I'd be happy to be proven wrong in case there is a better way to leverage Quasar's features. 

I could not find any docs or examples of handing this `<picture>` use case with Quasar, so would be happy to see any comments.

**Comment by ivanjaros** [comment]
https://quasar.dev/vue-components/img#responsive ?

**Reply by Divensky** [reply]
Thanks for the comment! However this link takes care of "[resolution switching](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images#resolution_switching_different_sizes)": displaying one image while providing the browser with a choice of its sizes or resolutions: `image-1x`, `image-2x`, `image-3x`... What I am looking for is called "[art direction](https://web.dev/learn/images/prescriptive#art_directed_images_with_picture)" and involves serving different images at different breakpoints.
